clouddriver-openstack/src/test/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/view/OpenstackSecurityGroupProviderSpec.groovy

```diff
@@ -148,6 +154,51 @@ class OpenstackSecurityGroupProviderSpec extends Specification {
     'account2' | 'west' | 'name-west' | allSecurityGroups | account2West[0]
   }
 
+  def "get all with rules"() {
+    given:
+    String region = 'region'
+    SecGroupExtension secGroupExtension = new NovaSecGroupExtension(id: UUID.randomUUID().toString(),
+      name: 'sec-group',
+      description: "description",
+      rules: [
+        createRule(1000, 1003, '10.10.0.0/32', IPProtocol.TCP),
+        createRule(2000, 2003, '0.0.0.0/24', IPProtocol.TCP),
+        createRule(80, 80, '192.168.1.1', IPProtocol.TCP)
+      ]
+    )
+    OpenstackClientProvider clientProvider = Mock()
+    GroovyMock(OpenstackProviderFactory, global: true)
+    OpenstackNamedAccountCredentials namedAccountCredentials = Mock()
+    OpenstackProviderFactory.createProvider(namedAccountCredentials) >> { clientProvider }
+    OpenstackCredentials credentials = new OpenstackCredentials(namedAccountCredentials)
+
+    when:
+    Set<OpenstackSecurityGroup> securityGroups = provider.getAll(true)
+
+    then:
+    1 * accountCredentialsProvider.getAll() >> [namedAccountCredentials]
+    1 * namedAccountCredentials.getCredentials() >> credentials
+    _ * namedAccountCredentials.name >> 'os-account'
+    1 * clientProvider.getAllRegions() >> [region]
+    1 * clientProvider.getSecurityGroups(region) >> [secGroupExtension]
+
+    and:
```

Comment:
Might be stylistic, but does this look better?
``
securityGroups.first().with {
  accountName == 'os-account'
  ...
}
``